contracts/test/ocr2/sequencer-uptime-feed.ts

```diff
@@ -0,0 +1,121 @@
+import { expect } from 'chai'
+import { starknet } from 'hardhat'
+import { StarknetContract, StarknetContractFactory, Account } from 'hardhat/types/runtime'
+import { number } from 'starknet'
+import { assertErrorMsg } from '../utils'
+
+describe('SequencerUptimeFeed test', function () {
+  this.timeout(300_000)
+
+  let account: Account
+  let nonOwner: Account
+
+  // should be beforeeach, but that'd be horribly slow. Just remember that the tests are not idempotent
+  before(async function () {
+    account = await starknet.deployAccount('OpenZeppelin')
+    nonOwner = await starknet.deployAccount('OpenZeppelin')
+  })
+
+  describe('Test inheritence', function () {
+    const user = 101
+    let uptimeFeedContract: StarknetContract
+
+    before(async function () {
+      const uptimeFeedFactory = await starknet.getContractFactory('sequencer_uptime_feed')
+      uptimeFeedContract = await uptimeFeedFactory.deploy({
+        initial_status: 0,
+        owner_address: number.toBN(account.starknetContract.address),
+      })
+    })
+
+    it('Test grainting access', async function () {
```

Comment:
We should structure our tests where there is a describe block for each function and have describe blocks for each scenario.  This is an example.

```
describe('#add_access', () => {
  describe('when called by non owner', () => {
    it('reverts', async () => {
       // Add test here
    })
  })
})
```

Reply:
Let's create an extra PR for test refactorings

Reply:
We'll add this in the same PR after all :)